src/main/java/io/kestra/plugin/perplexity/ChatCompletion.java

```diff
@@ -0,0 +1,214 @@
+    )
+    @Builder.Default
+    private Property<@Max(2) Double> frequencyPenalty = Property.ofValue(0.0);
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        var renderedApiKey = runContext.render(this.apiKey).as(String.class).orElseThrow();
+        var renderedModel = runContext.render(this.model).as(String.class).orElseThrow();
+        var renderedTemperature = runContext.render(this.temperature).as(Double.class).orElse(0.2);
+        var renderedTopP = runContext.render(this.topP).as(Double.class).orElse(0.9);
+        var renderedTopK = runContext.render(this.topK).as(Integer.class).orElse(50);
+        var renderedStream = runContext.render(this.stream).as(Boolean.class).orElse(false);
+        var renderedPresencePenalty = runContext.render(this.presencePenalty).as(Double.class).orElse(0.0);
+        var renderedFrequencyPenalty = runContext.render(this.frequencyPenalty).as(Double.class).orElse(0.0);
+
+        var renderedMessages = runContext.render(this.messages)
+            .asList(ChatMessage.class)
+            .stream()
+            .map(msg -> Map.of("role", msg.type().role(), "content", Objects.toString(msg.content(), "")))
+            .toList();
+
+        var requestBody = Map.of(
+            "model", renderedModel,
+            "messages", renderedMessages,
+            "temperature", renderedTemperature,
+            "top_p", renderedTopP,
+            "top_k", renderedTopK,
+            "stream", renderedStream,
+            "presence_penalty", renderedPresencePenalty,
+            "frequency_penalty", renderedFrequencyPenalty);
+
+        try (var client = new HttpClient(runContext, HttpConfiguration.builder().build())) {
+            var httpRequest = HttpRequest.builder()
+                .uri(URI.create(API_URL))
+                .method("POST")
+                .addHeader("Authorization", "Bearer " + renderedApiKey)
+                .addHeader("Content-Type", "application/json")
+                .body(HttpRequest.JsonRequestBody.builder()
+                    .content(requestBody)
+                    .build())
+                .build();
+
+            HttpResponse<String> response = client.request(httpRequest, String.class);
+
+            if (response.getStatus().getCode() >= 400) {
+                throw new IOException("Perplexity API error: " + response.getBody());
+            }
+
+            var parsed = JacksonMapper.ofJson().readValue(response.getBody(), Map.class);
+
+            var usage = (Map<String, Object>) parsed.get("usage");
+            if (usage != null) {
+                runContext.metric(Counter.of("usage.prompt.tokens", ((Number) usage.get("prompt_tokens")).longValue()));
+                runContext.metric(Counter.of("usage.completion.tokens", ((Number) usage.get("completion_tokens")).longValue()));
+                runContext.metric(Counter.of("usage.total.tokens", ((Number) usage.get("total_tokens")).longValue()));
```

Comment:
Could be great to have also `maxTokens` as a property ([see doc](https://docs.perplexity.ai/api-reference/chat-completions-post#body-max-tokens))!